client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -275,46 +268,110 @@ export function AdminMatchmakingTimes() {
 
       <PageHeadline>Matchmaking times</PageHeadline>
 
-      {/** Recreate this component when the active tab changes so it clears its local state. */}
-      <AddAndDisplayMatchmakingTimes key={activeTab} activeTab={activeTab} />
+      {/**
+       * Recreate this component when the active tab changes so it clears its local state (and the
+       * local state of its children components).
+       */}
+      <HistoryContainer key={activeTab}>
+        <FutureMatchmakingTimes activeTab={activeTab} />
+        <CurrentMatchmakingTime activeTab={activeTab} />
+        <PastMatchmakingTimes activeTab={activeTab} />
+      </HistoryContainer>
     </Container>
   )
 }
 
-function AddAndDisplayMatchmakingTimes({ activeTab }: { activeTab: MatchmakingType }) {
+function FutureMatchmakingTimes({ activeTab }: { activeTab: MatchmakingType }) {
   const dispatch = useAppDispatch()
   const snackbarController = useSnackbarController()
 
   const [startDate, setStartDate] = useState('')
   const [invalidDate, setInvalidDate] = useState(false)
   const [enabled, setEnabled] = useState(false)
-
-  const [error, setError] = useState<Error>()
+  const [addingTimeError, setAddingTimeError] = useState<Error>()
 
   const [futureTimes, setFutureTimes] = useState<MatchmakingTimeJson[]>([])
+  const [hasMoreFutureTimes, setHasMoreFutureTimes] = useState(true)
+  const [isLoadingFutureTimes, setIsLoadingFutureTimes] = useState(false)
+  const [loadingFutureTimesError, setLoadingFutureTimesError] = useState<Error | undefined>(
+    undefined,
+  )
+
+  const futureTimesRef = useValueAsRef(futureTimes)
```

Comment:
I dunno if this is deprecated or not but it should be, so please stop writing new code using it. To accomplish what you're doing here without it, just use the callback form of the state setter. Apply throughout